UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/ModelRunner.cs

```diff
@@ -106,6 +109,14 @@ public void PutObservations(string key, ICollection<Agent> agents)
                 return;
             }
 
+            if (!m_visualObservationsInitialized)
+            {
+                // Just grab the first agent in the collection (any will suffice, really).
+                var firstAgent = agents.FirstOrDefault();
```

Comment:
Is it possible for FirstOrDefault to return a default (an agent that does not exist) and mess up the Initialization of the TensorGenerator ?

Reply:
A few lines above, we early out if `agents.Count` is 0, so we should always have a valid Agent to grab here. I'll change this to `First()` and a comment though, so that it's clearer.